pkg/internal/export/grafana_agent/traces.go

```diff
@@ -0,0 +1,119 @@
+package grafana_agent
+
+import (
+	"context"
+	"time"
+
+	"github.com/mariomac/pipes/pkg/node"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+
+	"github.com/grafana/beyla/pkg/beyla"
+	"github.com/grafana/beyla/pkg/internal/export/otel"
+	"github.com/grafana/beyla/pkg/internal/request"
+)
+
+// TracesReceiver creates a terminal node that consumes request.Spans and sends OpenTelemetry traces to the configured consumers.
+func TracesReceiver(ctx context.Context, cfg beyla.TracesReceiverConfig) (node.TerminalFunc[[]request.Span], error) {
+	return func(in <-chan []request.Span) {
+		for spans := range in {
+			for _, span := range spans {
+				if span.IgnoreSpan == request.IgnoreTraces {
+					continue
+				}
+
+				t := span.Timings()
+				parentCtx := otel.HandleTraceparent(ctx, &span)
+				realStart := otel.SpanStartTime(t)
+				hasSubSpans := t.Start.After(realStart)
+				if !hasSubSpans {
+					// We set the eBPF calculated trace_id and span_id to be the main span
+					parentCtx = otel.ContextWithTraceParent(parentCtx, span.TraceID, span.SpanID)
```

Comment:
@mariomac In the original implementation of traces exporter, we do this trick. I think in the case of using collector methods, maybe this won't do anything and we need to set those fields (`TraceID` and `SpanID` manually) but I'm not sure if it's necessary.

Reply:
Yeah, neither I am sure. Maybe @grcevski can put some light into this.

Reply:
We use this OTel IdGenerator logic, which injects the traceID and spanID we have produced from the eBPF side. We started doing this when we had to implement context propagation. Essentially, the child span may start before the userspace side in Beyla will see the event of the parent span, so we needed to know the TraceID in eBPF code.

The OTel API is not very amenable to changing the TraceID and the SpanID, the only way I was able to do that is with that custom IdGenerator, which takes the eBPF ids if they exist or makes new ones if they were empty.

The sub-spans essentially means that we have the in-queue and the processing span. 

1. If there's no sub-spans, the SpanID generated from eBPF code becomes the SpanID of the main span.
2. If there are sub-spans, we generate a new SpanID for the main span of the server request, make that the parent of the eBPF generated SpanID and we set the eBPF SpanID on the "processing" span.

This way if there are no sub-spans, the trace looks like this:

```
HTTP server (SpanID generated by eBPF)
    HTTP client (SpanID generated by eBPF)
```

if there are in-queue and processing sub-spans, it looks like this:

```
HTTP server (SpanID generated by Go user space IdGenerator)
    in-queue (SpanID generated by Go user space IdGenerator)
    processing (SpanID generated by eBPF) 
         HTTP client (SpanID generated by eBPF)
```

TraceID is always generated by eBPF.